Бисултанов У.З. Лаб.4 Группа 6312

UmarBis opened the pull request:
**ФИО:** Бисултанов Умар 
**Номер группы:** 6312 
**Номер лабораторной:** 4
**Краткое описание исходного кода:** Я взял исходный код из своей старой практической работы library_before.py
**Краткое описание проделанной работы:** Я сделал рефакторинг (library_final.py).

sk1fix commented:
`import sys, os, re, time
import gzip, getopt, socket
`
Все импорты должны быть на отдельных строках

sk1fix reviewed (commented):
Для улучшения читабельности и поддерживаемости кода необходимо:

1)Привести код к PEP8:
2)Документирование
3)Устранение дублирования:
4)Совместимость с Python 3
5)Рефакторинг структуры:

Эти изменения улучшат читаемость, производительность и адаптируемость кода.

alxmcs commented:
Зачтено

UmarBis committed:
Create library_before.py

Проблемы:

Нарушение принципа единственной ответственности - один класс делает всё

Сильная связанность - все компоненты тесно переплетены

Повторяющийся код - много дублирования в поисковых операциях

Смешение уровней абстракции - бизнес-логика смешана с UI

Жестко закодированные зависимости - сложно тестировать и расширять

Отсутствие инкапсуляции - прямой доступ к внутренним структурам данных

UmarBis committed:
Create library_final.py

UmarBis commented:
Улучшения после рефакторинга:

1. Принцип единственной ответственности
-Разделен на отдельные классы: репозитории, сервисы, модели
-Каждый класс отвечает за одну конкретную задачу

2. Инверсия зависимостей
3. Улучшенная архитектура

4. Устранение дублирования кода
-Общая логика поиска вынесена в репозитории
-Повторяющиеся операции инкапсулированы

5. Улучшенная расширяемость
-Легко добавить новые типы отчетов
-Просто интегрировать новые источники данных
-Можно добавить новые сервисы без изменения существующих

6. Лучшая сопровождаемость
-Четкая структура проекта
-Явные зависимости
-Легко находить и исправлять ошибки

alxmcs commented:
Зачтено
